services/mirror/mirror.go

```diff
@@ -46,14 +46,18 @@ func doMirrorSync(ctx context.Context, req *SyncRequest) {
 }
 
 // Update checks and updates mirror repositories.
-func Update(ctx context.Context) error {
+func Update(ctx context.Context, pullLimit, pushLimit int) error {
 	if !setting.Mirror.Enabled {
 		log.Warn("Mirror feature disabled, but cron job enabled: skip update")
 		return nil
 	}
 	log.Trace("Doing: Update")
 
-	handler := func(idx int, bean interface{}) error {
+	requested := 0
+
+	doneErr := fmt.Errorf("reached limit")
```

Comment:
Why exactly is this error always constructed, and not only when the limit has been reached?

Reply:
because it is later checked against for equality 

See line 122:

```go
	}); err != nil && err != doneErr {
```

However, I will stop repeatedly creating this and instead just create it once.